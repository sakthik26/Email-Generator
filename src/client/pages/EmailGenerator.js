import React from 'react';
import Avatar from '@material-ui/core/Avatar';
import Button from '@material-ui/core/Button';
import CssBaseline from '@material-ui/core/CssBaseline';
import TextField from '@material-ui/core/TextField';
import FormControlLabel from '@material-ui/core/FormControlLabel';
import Checkbox from '@material-ui/core/Checkbox';
import Link from '@material-ui/core/Link';
import Grid from '@material-ui/core/Grid';
import Box from '@material-ui/core/Box';
import Typography from '@material-ui/core/Typography';
import { makeStyles } from '@material-ui/core/styles';
import Container from '@material-ui/core/Container';
import { useEffect, useState } from "react";
import axios from 'axios';
import InputLabel from '@material-ui/core/InputLabel';
import MenuItem from '@material-ui/core/MenuItem';
import FormHelperText from '@material-ui/core/FormHelperText';
import FormControl from '@material-ui/core/FormControl';
import Select from '@material-ui/core/Select';
import Alert from '@material-ui/lab/Alert';
import IconButton from '@material-ui/core/IconButton';
import CloseIcon from '@material-ui/icons/Close';
import Collapse from '@material-ui/core/Collapse';

const useStyles = makeStyles((theme) => ({
    root: {
        width: '100%',
        '& > * + *': {
            marginTop: theme.spacing(2),
        },
    },
    formControl: {
        margin: theme.spacing(1),
        minWidth: 380,
    },
    selectEmpty: {
        marginTop: theme.spacing(2),
    },
    paper: {
        marginTop: theme.spacing(8),
        display: 'flex',
        flexDirection: 'column',
        alignItems: 'center',
    },
    avatar: {
        margin: theme.spacing(1),
        backgroundColor: theme.palette.secondary.main,
    },
    form: {
        width: '100%', // Fix IE 11 issue.
        marginTop: theme.spacing(3),
    },
    submit: {
        margin: theme.spacing(3, 0, 2),
    },
    spacing: {
        marginTop: 20,
    },
    cusor: {
        cursor: 'pointer'
    }
}));

export default function EmailGenerator() {
    const classes = useStyles();
    const [open, setOpen] = React.useState(true);
    const [openErrorAlert, setOpenErrorAlert] = React.useState(true);
    const [companyDomainOptions, setCompanyDomainOptions] = React.useState([

    ])
    const [showGenerate, setShowGenerate] = React.useState(false);
    const [generatedEmail, setGeneratedEmail] = React.useState('');
    const [firstName, setFirstName] = React.useState('');
    const [success, setSuccess] = React.useState(false)
    const [failed, setFailed] = React.useState(false)
    const [lastName, setLastName] = React.useState('');
    const [createFirstName, setCreateFirstName] = React.useState('');
    const [createLastName, setCreateLastName] = React.useState('');
    const [createEmailAddress, setCreateEmailAddress] = React.useState('');
    const [domainSelected, setDomainSelected] = React.useState('')
    const handleChange = (event) => {
        console.log(event.target)
        setDomainSelected(event.target.value);
    };
    const changeFirstName = (e) => {
        setFirstName(e.target.value)
    }

    const changeLastName = (e) => {
        setLastName(e.target.value)
    }
    const changeCreateFirstName = (e) => {
        setCreateFirstName(e.target.value)
    }
    const changeCreateLastName = (e) => {
        setCreateLastName(e.target.value)
    }
    const changeEmail = (e) => {
        setCreateEmailAddress(e.target.value)
    }
    useEffect(() => {
        //Fetches all the domains of the company from the database
        axios
            .get("/api/getdomains", {
            })
            .then((response) => {

                setCompanyDomainOptions(response.data);
                setDomainSelected(response.data[0].domain)

            })
            .catch(function (e) {
                console.log(e);
            });

    }, []);

    // Generates the email address based on previous address of the domain
    const generateEmailAddress = (event) => {

        const payload = {
            firstname: firstName,
            lastname: lastName,
            domain: domainSelected
        }

        axios.post("/api/generatemail", payload, {
            headers: {
                "Content-type": "application/json"
            }
        })
            .then((response) => {
                setGeneratedEmail(response.data.data)
            })
            .catch(function (e) {
                console.log(e);
            });
    }

    //Creates new domain if the domain doesn't exist and reports if the domain already exists
    const createDomain = (event) => {

        const payload = {
            firstname: createFirstName,
            lastname: createLastName,
            email: createEmailAddress
        }
        axios.post("/api/createdomain", payload, {
            headers: {
                "Content-type": "application/json"
            }
        })
            .then((response) => {
                if (response.data.data === "Domain already exists") {
                    setFailed(true)
                    setSuccess(false)
                }
                else {
                    setCompanyDomainOptions([...companyDomainOptions, response.data.data])
                    setSuccess(true)
                    setFailed(false)
                }
            })
            .catch(function (e) {
                console.log(e);
            });
    }
    return (
        <Container component="main" maxWidth="xs">
            <CssBaseline />
            <div className={classes.paper}>
                <Avatar className={classes.avatar}>

                </Avatar>
                <Typography component="h1" variant="h5">
                    Email Generator
                </Typography>
                <form className={classes.form} noValidate>
                    <Grid container spacing={2}>
                        <Grid item xs={12} sm={6}>
                            <TextField
                                autoComplete="fname"
                                name="firstName"
                                variant="outlined"
                                required
                                onChange={changeFirstName}
                                value={firstName}
                                fullWidth
                                id="firstName"
                                label="First Name"
                                autoFocus
                            />
                        </Grid>
                        <Grid item xs={12} sm={6}>
                            <TextField
                                variant="outlined"
                                required
                                value={lastName}
                                fullWidth
                                id="lastName"
                                onChange={changeLastName}
                                label="Last Name"
                                name="lastName"
                                autoComplete="lname"
                            />
                        </Grid>
                        <Grid item xs={12}>
                            <FormControl className={classes.formControl}>
                                <InputLabel id="demo-simple-select-label">Domain name</InputLabel>
                                <Select
                                    labelId="demo-simple-select-label"
                                    id="demo-simple-select"
                                    value={domainSelected}
                                    onChange={handleChange}
                                >
                                    {companyDomainOptions.map(item => (
                                        <MenuItem
                                            value={item.domain}
                                        >
                                            {item.domain}
                                        </MenuItem>
                                    ))}

                                </Select>
                            </FormControl>
                        </Grid>


                    </Grid>
                    <Button
                        fullWidth
                        variant="contained"
                        color="primary"
                        className={classes.submit}
                        onClick={generateEmailAddress}
                    >
                        Generate Email address
                    </Button>
                    <Grid container spacing={2}>
                        <Grid item xs={12}>
                            <TextField
                                id="outlined-read-only-input"
                                InputProps={{
                                    readOnly: true,
                                }}
                                fullWidth
                                label="Email address"
                                value={generatedEmail}
                                defaultValue="Email appears here"
                                variant="outlined"
                            />
                        </Grid>
                    </Grid>
                    <Grid container>
                        <Grid item className={classes.spacing}>
                            <Link onClick={() => setShowGenerate(true)} variant="body2" className={classes.cusor} >
                                Can't find the company domain?
                            </Link>
                        </Grid>
                        {showGenerate ?
                            <div>
                                <Grid item xs={12} className={classes.spacing}>
                                    <TextField
                                        autoComplete="fname"
                                        name="firstName"
                                        variant="outlined"
                                        required
                                        onChange={changeCreateFirstName}
                                        value={createFirstName}
                                        fullWidth
                                        id="firstName"
                                        label="First Name"
                                        autoFocus
                                    />
                                </Grid>
                                <Grid item xs={12} className={classes.spacing}>
                                    <TextField
                                        variant="outlined"
                                        required
                                        value={createLastName}
                                        fullWidth
                                        id="lastName"
                                        onChange={changeCreateLastName}
                                        label="Last Name"
                                        name="lastName"
                                        autoComplete="lname"
                                    />
                                </Grid>

                                <Grid item xs={12} className={classes.spacing}>
                                    <TextField
                                        autoComplete="email"
                                        name="email"
                                        variant="outlined"
                                        required
                                        onChange={changeEmail}
                                        value={createEmailAddress}
                                        fullWidth
                                        id="firstName"
                                        label="Enter Email"
                                        autoFocus
                                    />
                                </Grid>
                                <Button
                                    fullWidth
                                    variant="contained"
                                    color="primary"
                                    className={classes.submit}
                                    onClick={createDomain}
                                >
                                    Create Company Domain Name
                                </Button>
                            </div> : null}
                    </Grid>
                </form>
            </div >
            {success ?
                <div className={classes.root}>
                    <Collapse in={open}>
                        <Alert
                            action={
                                <IconButton
                                    aria-label="close"
                                    color="inherit"
                                    size="small"
                                    onClick={() => {
                                        setOpen(false);
                                    }}
                                >
                                    <CloseIcon fontSize="inherit" />
                                </IconButton>
                            }
                        >
                            Domain Created Successfully
                        </Alert>
                    </Collapse>
                </div> : null}

            {failed ?
                <div className={classes.root}>
                    <Collapse in={openErrorAlert}>
                        <Alert severity="error"
                            action={
                                <IconButton
                                    aria-label="close"
                                    color="inherit"
                                    size="small"
                                    onClick={() => {
                                        setOpenErrorAlert(false);
                                    }}
                                >
                                    <CloseIcon fontSize="inherit" />
                                </IconButton>
                            }
                        >
                            Domain already exists. Please choose another domain.
                        </Alert>
                    </Collapse>
                </div> : null}

        </Container >
    );
}