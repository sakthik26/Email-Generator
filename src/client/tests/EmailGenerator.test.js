import React from 'react';
import renderer from 'react-test-renderer';
import EmailGenerator from '../pages/EmailGenerator.js'

test('Match the snapshot of the Email Generator', () => {
    const component = renderer.create(
        <EmailGenerator />
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();

});