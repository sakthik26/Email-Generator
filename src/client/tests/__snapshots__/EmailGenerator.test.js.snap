// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Link changes the class when hovered 1`] = `
<main
  className="MuiContainer-root MuiContainer-maxWidthXs"
>
  <div
    className="makeStyles-paper-4"
  >
    <div
      className="MuiAvatar-root MuiAvatar-circle makeStyles-avatar-5 MuiAvatar-colorDefault"
    >
      <svg
        aria-hidden={true}
        className="MuiSvgIcon-root MuiAvatar-fallback"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M12 12c2.21 0 4-1.79 4-4s-1.79-4-4-4-4 1.79-4 4 1.79 4 4 4zm0 2c-2.67 0-8 1.34-8 4v2h16v-2c0-2.66-5.33-4-8-4z"
        />
      </svg>
    </div>
    <h1
      className="MuiTypography-root MuiTypography-h5"
    >
      Email Generator
    </h1>
    <form
      className="makeStyles-form-6"
      noValidate={true}
    >
      <div
        className="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-2"
      >
        <div
          className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-6"
        >
          <div
            className="MuiFormControl-root MuiTextField-root MuiFormControl-fullWidth"
          >
            <label
              className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined Mui-required Mui-required"
              data-shrink={false}
              htmlFor="firstName"
              id="firstName-label"
            >
              First Name
              <span
                aria-hidden={true}
                className="MuiFormLabel-asterisk MuiInputLabel-asterisk"
              >
                 
                *
              </span>
            </label>
            <div
              className="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-fullWidth MuiInputBase-formControl"
              onClick={[Function]}
            >
              <input
                aria-invalid={false}
                autoComplete="fname"
                autoFocus={true}
                className="MuiInputBase-input MuiOutlinedInput-input"
                disabled={false}
                id="firstName"
                name="firstName"
                onAnimationStart={[Function]}
                onBlur={[Function]}
                onChange={[Function]}
                onFocus={[Function]}
                required={true}
                type="text"
                value=""
              />
              <fieldset
                aria-hidden={true}
                className="PrivateNotchedOutline-root-10 MuiOutlinedInput-notchedOutline"
              >
                <legend
                  className="PrivateNotchedOutline-legendLabelled-12"
                >
                  <span>
                    First Name
                     *
                  </span>
                </legend>
              </fieldset>
            </div>
          </div>
        </div>
        <div
          className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-6"
        >
          <div
            className="MuiFormControl-root MuiTextField-root MuiFormControl-fullWidth"
          >
            <label
              className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined Mui-required Mui-required"
              data-shrink={false}
              htmlFor="lastName"
              id="lastName-label"
            >
              Last Name
              <span
                aria-hidden={true}
                className="MuiFormLabel-asterisk MuiInputLabel-asterisk"
              >
                 
                *
              </span>
            </label>
            <div
              className="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-fullWidth MuiInputBase-formControl"
              onClick={[Function]}
            >
              <input
                aria-invalid={false}
                autoComplete="lname"
                autoFocus={false}
                className="MuiInputBase-input MuiOutlinedInput-input"
                disabled={false}
                id="lastName"
                name="lastName"
                onAnimationStart={[Function]}
                onBlur={[Function]}
                onChange={[Function]}
                onFocus={[Function]}
                required={true}
                type="text"
                value=""
              />
              <fieldset
                aria-hidden={true}
                className="PrivateNotchedOutline-root-10 MuiOutlinedInput-notchedOutline"
              >
                <legend
                  className="PrivateNotchedOutline-legendLabelled-12"
                >
                  <span>
                    Last Name
                     *
                  </span>
                </legend>
              </fieldset>
            </div>
          </div>
        </div>
        <div
          className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12"
        >
          <div
            className="MuiFormControl-root makeStyles-formControl-2"
          >
            <label
              className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated"
              data-shrink={false}
              id="demo-simple-select-label"
            >
              Domain name
            </label>
            <div
              className="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-formControl MuiInput-formControl"
              onClick={[Function]}
            >
              <div
                aria-haspopup="listbox"
                aria-labelledby="demo-simple-select-label demo-simple-select"
                className="MuiSelect-root MuiSelect-select MuiSelect-selectMenu MuiInputBase-input MuiInput-input"
                id="demo-simple-select"
                onBlur={[Function]}
                onFocus={[Function]}
                onKeyDown={[Function]}
                onMouseDown={[Function]}
                role="button"
                tabIndex={0}
              >
                <span
                  dangerouslySetInnerHTML={
                    Object {
                      "__html": "&#8203;",
                    }
                  }
                />
              </div>
              <input
                aria-hidden={true}
                className="MuiSelect-nativeInput"
                onAnimationStart={[Function]}
                onChange={[Function]}
                required={false}
                tabIndex={-1}
                value=""
              />
              <svg
                aria-hidden={true}
                className="MuiSvgIcon-root MuiSelect-icon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M7 10l5 5 5-5z"
                />
              </svg>
            </div>
          </div>
        </div>
      </div>
      <button
        className="MuiButtonBase-root MuiButton-root MuiButton-contained makeStyles-submit-7 MuiButton-containedPrimary MuiButton-fullWidth"
        disabled={false}
        onBlur={[Function]}
        onClick={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex={0}
        type="button"
      >
        <span
          className="MuiButton-label"
        >
          Generate Email address
        </span>
      </button>
      <div
        className="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-2"
      >
        <div
          className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12"
        >
          <div
            className="MuiFormControl-root MuiTextField-root MuiFormControl-fullWidth"
          >
            <label
              className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-outlined MuiFormLabel-filled"
              data-shrink={true}
              htmlFor="outlined-read-only-input"
              id="outlined-read-only-input-label"
            >
              Email address
            </label>
            <div
              className="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-fullWidth MuiInputBase-formControl"
              onClick={[Function]}
            >
              <input
                aria-invalid={false}
                autoFocus={false}
                className="MuiInputBase-input MuiOutlinedInput-input"
                defaultValue="Email appears here"
                disabled={false}
                id="outlined-read-only-input"
                onAnimationStart={[Function]}
                onBlur={[Function]}
                onChange={[Function]}
                onFocus={[Function]}
                readOnly={true}
                required={false}
                type="text"
                value=""
              />
              <fieldset
                aria-hidden={true}
                className="PrivateNotchedOutline-root-10 MuiOutlinedInput-notchedOutline"
              >
                <legend
                  className="PrivateNotchedOutline-legendLabelled-12 PrivateNotchedOutline-legendNotched-13"
                >
                  <span>
                    Email address
                  </span>
                </legend>
              </fieldset>
            </div>
          </div>
        </div>
      </div>
      <div
        className="MuiGrid-root MuiGrid-container"
      >
        <div
          className="MuiGrid-root makeStyles-spacing-8 MuiGrid-item"
        >
          <a
            className="MuiTypography-root MuiLink-root MuiLink-underlineHover makeStyles-cusor-9 MuiTypography-body2 MuiTypography-colorPrimary"
            onBlur={[Function]}
            onClick={[Function]}
            onFocus={[Function]}
          >
            Can't find the company domain?
          </a>
        </div>
      </div>
    </form>
  </div>
</main>
`;

exports[`Match the snapshot of the Email Generator 1`] = `
<main
  className="MuiContainer-root MuiContainer-maxWidthXs"
>
  <div
    className="makeStyles-paper-4"
  >
    <div
      className="MuiAvatar-root MuiAvatar-circle makeStyles-avatar-5 MuiAvatar-colorDefault"
    >
      <svg
        aria-hidden={true}
        className="MuiSvgIcon-root MuiAvatar-fallback"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M12 12c2.21 0 4-1.79 4-4s-1.79-4-4-4-4 1.79-4 4 1.79 4 4 4zm0 2c-2.67 0-8 1.34-8 4v2h16v-2c0-2.66-5.33-4-8-4z"
        />
      </svg>
    </div>
    <h1
      className="MuiTypography-root MuiTypography-h5"
    >
      Email Generator
    </h1>
    <form
      className="makeStyles-form-6"
      noValidate={true}
    >
      <div
        className="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-2"
      >
        <div
          className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-6"
        >
          <div
            className="MuiFormControl-root MuiTextField-root MuiFormControl-fullWidth"
          >
            <label
              className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined Mui-required Mui-required"
              data-shrink={false}
              htmlFor="firstName"
              id="firstName-label"
            >
              First Name
              <span
                aria-hidden={true}
                className="MuiFormLabel-asterisk MuiInputLabel-asterisk"
              >
                 
                *
              </span>
            </label>
            <div
              className="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-fullWidth MuiInputBase-formControl"
              onClick={[Function]}
            >
              <input
                aria-invalid={false}
                autoComplete="fname"
                autoFocus={true}
                className="MuiInputBase-input MuiOutlinedInput-input"
                disabled={false}
                id="firstName"
                name="firstName"
                onAnimationStart={[Function]}
                onBlur={[Function]}
                onChange={[Function]}
                onFocus={[Function]}
                required={true}
                type="text"
                value=""
              />
              <fieldset
                aria-hidden={true}
                className="PrivateNotchedOutline-root-10 MuiOutlinedInput-notchedOutline"
              >
                <legend
                  className="PrivateNotchedOutline-legendLabelled-12"
                >
                  <span>
                    First Name
                     *
                  </span>
                </legend>
              </fieldset>
            </div>
          </div>
        </div>
        <div
          className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-6"
        >
          <div
            className="MuiFormControl-root MuiTextField-root MuiFormControl-fullWidth"
          >
            <label
              className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined Mui-required Mui-required"
              data-shrink={false}
              htmlFor="lastName"
              id="lastName-label"
            >
              Last Name
              <span
                aria-hidden={true}
                className="MuiFormLabel-asterisk MuiInputLabel-asterisk"
              >
                 
                *
              </span>
            </label>
            <div
              className="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-fullWidth MuiInputBase-formControl"
              onClick={[Function]}
            >
              <input
                aria-invalid={false}
                autoComplete="lname"
                autoFocus={false}
                className="MuiInputBase-input MuiOutlinedInput-input"
                disabled={false}
                id="lastName"
                name="lastName"
                onAnimationStart={[Function]}
                onBlur={[Function]}
                onChange={[Function]}
                onFocus={[Function]}
                required={true}
                type="text"
                value=""
              />
              <fieldset
                aria-hidden={true}
                className="PrivateNotchedOutline-root-10 MuiOutlinedInput-notchedOutline"
              >
                <legend
                  className="PrivateNotchedOutline-legendLabelled-12"
                >
                  <span>
                    Last Name
                     *
                  </span>
                </legend>
              </fieldset>
            </div>
          </div>
        </div>
        <div
          className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12"
        >
          <div
            className="MuiFormControl-root makeStyles-formControl-2"
          >
            <label
              className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated"
              data-shrink={false}
              id="demo-simple-select-label"
            >
              Domain name
            </label>
            <div
              className="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-formControl MuiInput-formControl"
              onClick={[Function]}
            >
              <div
                aria-haspopup="listbox"
                aria-labelledby="demo-simple-select-label demo-simple-select"
                className="MuiSelect-root MuiSelect-select MuiSelect-selectMenu MuiInputBase-input MuiInput-input"
                id="demo-simple-select"
                onBlur={[Function]}
                onFocus={[Function]}
                onKeyDown={[Function]}
                onMouseDown={[Function]}
                role="button"
                tabIndex={0}
              >
                <span
                  dangerouslySetInnerHTML={
                    Object {
                      "__html": "&#8203;",
                    }
                  }
                />
              </div>
              <input
                aria-hidden={true}
                className="MuiSelect-nativeInput"
                onAnimationStart={[Function]}
                onChange={[Function]}
                required={false}
                tabIndex={-1}
                value=""
              />
              <svg
                aria-hidden={true}
                className="MuiSvgIcon-root MuiSelect-icon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M7 10l5 5 5-5z"
                />
              </svg>
            </div>
          </div>
        </div>
      </div>
      <button
        className="MuiButtonBase-root MuiButton-root MuiButton-contained makeStyles-submit-7 MuiButton-containedPrimary MuiButton-fullWidth"
        disabled={false}
        onBlur={[Function]}
        onClick={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex={0}
        type="button"
      >
        <span
          className="MuiButton-label"
        >
          Generate Email address
        </span>
      </button>
      <div
        className="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-2"
      >
        <div
          className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12"
        >
          <div
            className="MuiFormControl-root MuiTextField-root MuiFormControl-fullWidth"
          >
            <label
              className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-outlined MuiFormLabel-filled"
              data-shrink={true}
              htmlFor="outlined-read-only-input"
              id="outlined-read-only-input-label"
            >
              Email address
            </label>
            <div
              className="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-fullWidth MuiInputBase-formControl"
              onClick={[Function]}
            >
              <input
                aria-invalid={false}
                autoFocus={false}
                className="MuiInputBase-input MuiOutlinedInput-input"
                defaultValue="Email appears here"
                disabled={false}
                id="outlined-read-only-input"
                onAnimationStart={[Function]}
                onBlur={[Function]}
                onChange={[Function]}
                onFocus={[Function]}
                readOnly={true}
                required={false}
                type="text"
                value=""
              />
              <fieldset
                aria-hidden={true}
                className="PrivateNotchedOutline-root-10 MuiOutlinedInput-notchedOutline"
              >
                <legend
                  className="PrivateNotchedOutline-legendLabelled-12 PrivateNotchedOutline-legendNotched-13"
                >
                  <span>
                    Email address
                  </span>
                </legend>
              </fieldset>
            </div>
          </div>
        </div>
      </div>
      <div
        className="MuiGrid-root MuiGrid-container"
      >
        <div
          className="MuiGrid-root makeStyles-spacing-8 MuiGrid-item"
        >
          <a
            className="MuiTypography-root MuiLink-root MuiLink-underlineHover makeStyles-cusor-9 MuiTypography-body2 MuiTypography-colorPrimary"
            onBlur={[Function]}
            onClick={[Function]}
            onFocus={[Function]}
          >
            Can't find the company domain?
          </a>
        </div>
      </div>
    </form>
  </div>
</main>
`;
